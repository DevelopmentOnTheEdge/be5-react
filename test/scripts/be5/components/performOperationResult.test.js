import React          from 'react';
import renderer       from 'react-test-renderer';
import Document       from '../../../../src/scripts/be5/components/Document';
import be5            from '../../../../src/scripts/be5/be5';
import formService    from '../../../../src/scripts/be5/services/Forms';

test('performOperationResult finished', () => {
  const component = renderer.create(
    <Document frontendParams={{documentName: "test"}}/>
  );
  //const mockFunc = jest.fn();
  const res = {
    data: {
      type: "operationResult",
      attributes: {"status":"finished"}
    },
    links: {"self":"form/categories/Doc categories/Edit"},
    meta: {"_ts_":"1503244989281"}
  };

  formService._performOperationResult(res, {documentName: "test"});

  expect(component.toJSON()).toMatchSnapshot();
  //expect(mockFunc.mock.calls.length).toBe(1);
});

test('performOperationResult redirect', () => {
  //const mockFunc = jest.fn();
  be5.url.process = jest.fn();

  const res = {
    "data":{
      "type":"operationResult",
      "attributes":{"details":"static/welcome.be","status":"redirect"}
    },
    "links":{"self":"form/categories/Doc categories/Edit"},
    "meta":{"_ts_":"1503244989281"}
  };
  formService._performOperationResult(res, {documentName: "test"});

  expect(be5.url.process.mock.calls.length).toBe(1);
  //expect(mockFunc.mock.calls.length).toBe(1);
});

test('performOperationResult redirect main', () => {
  //const mockFunc = jest.fn();
  be5.url.set = jest.fn();

  const res = {
    "data":{
      "type":"operationResult",
      "attributes":{"details":"static/welcome.be","status":"redirect"}
    },
    "links":{"self":"form/categories/Doc categories/Edit"},
    "meta":{"_ts_":"1503244989281"}
  };
  formService._performOperationResult(res, {documentName: be5.mainDocumentName});

  expect(be5.url.set.mock.calls.length).toBe(1);
  //expect(mockFunc.mock.calls.length).toBe(1);
});
