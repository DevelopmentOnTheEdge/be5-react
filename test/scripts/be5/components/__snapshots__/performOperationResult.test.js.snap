// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performOperationResult finished 1`] = `
<div
  className="document-content"
  id="document-content___test"
>
  <div>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Successfully completed.",
        }
      }
    />
  </div>
</div>
`;
